lib/rucio/core/replica.py

```diff
@@ -101,16 +100,25 @@ def get_bad_replicas_summary(rse_expression=None, from_date=None, to_date=None,
         to_days = func.date_trunc('day', models.BadReplicas.created_at)
     else:
         to_days = func.strftime(models.BadReplicas.created_at, '%Y-%m-%d')
-    query = session.query(func.count(), to_days, models.BadReplicas.rse_id, models.BadReplicas.state, models.BadReplicas.reason)
+
+    stmt = select(
+        func.count(),
+        to_days,
+        models.BadReplicas.rse_id,
+        models.BadReplicas.state,
+        models.BadReplicas.reason
+    ).select_from(
+        models.BadReplicas
+    )
     # To be added : HINTS
     if rse_clause != []:
-        query = query.filter(or_(*rse_clause))
+        stmt = stmt.where(or_(*rse_clause))
     if from_date:
-        query = query.filter(models.BadReplicas.created_at > from_date)
+        stmt = stmt.where(models.BadReplicas.created_at > from_date)
     if to_date:
-        query = query.filter(models.BadReplicas.created_at < to_date)
-    summary = query.group_by(to_days, models.BadReplicas.rse_id, models.BadReplicas.reason, models.BadReplicas.state).all()
-    for row in summary:
+        stmt = stmt.where(models.BadReplicas.created_at < to_date)
+    stmt = stmt.group_by(to_days, models.BadReplicas.rse_id, models.BadReplicas.reason, models.BadReplicas.state)
+    for row in session.execute(stmt):
```

Comment:
Would be more readable to use ` for count, to_days, rse_id, reason in query:` instead of indexing the row.

Reply:
Hi, thanks for the thorough review and sorry for leaving this until now.

Changed to:
```
    for count, to_days, rse_id, state, reason in session.execute(stmt):
```